JumboCode/TheLegacyProject2
# Nickbar01234/file UI

# Description

Implement a generic component to display files for a given senior. The component can optionally allow users to change student assignment for a given senior. Note that this option should only be used in `CHAPTER_LEADER` view.

## Screenshots

Regular user / admin view

![image](https://github.com/JumboCode/TheLegacyProject2/assets/74647679/f95469ab-7cd2-4429-ad4b-c31d0d0e2a9a)

Chapter leader view

![image](https://github.com/JumboCode/TheLegacyProject2/assets/74647679/9d1562d8-a531-4102-9087-751d0dee5f55)

![image](https://github.com/JumboCode/TheLegacyProject2/assets/74647679/9364a314-5194-45f7-9273-77bed716c1c3)


## Tests

In `src/app/private/[uid]/user/seniors/[seniorId]/page.tsx`, update props to be `<DislaySenior editable senior={senior} />`.

1. Navigate to `/private/user/[userId]/seniors/[seniorId]`
2. Update role on database as `CHAPTER_LEADER`
3. Edit student assignments and verify on database


## Additional Documentation

This is a long PR, but a good amount of change came from a different ticket. I wanted get the changes in first so @Tia-Chen  and @nathan-j-edwards can implement the file popup. The core logic of the PR is in this [commit](https://github.com/JumboCode/TheLegacyProject2/pull/95/commits/a50cea4e9821b9446faeb64258f486d8d36b9f79).
